core/src/main/java/hudson/FilePath.java

```diff
@@ -458,13 +455,10 @@ public int archive(final ArchiverFactory factory, OutputStream os, final DirScan
         final OutputStream out = (channel!=null)?new RemoteOutputStream(os):os;
         return act(new SecureFileCallable<Integer>() {
             public Integer invoke(File f, VirtualChannel channel) throws IOException {
-                Archiver a = factory.create(out);
-                try {
+                try (Archiver a = factory.create(out)) {
                     scanner.scan(f,reading(a));
-                } finally {
-                    a.close();
+                    return a.countEntries();
```

Comment:
Why have you moved it into try block? now we count something (potentially long) before closing the archiver. 🐜